File: apps/commerce-api/src/main/java/com/loopers/application/payment/PaymentCommand.java
package com.loopers.application.payment;

import com.loopers.domain.payment.CardType;
import com.loopers.domain.payment.PaymentData;
import com.loopers.domain.payment.TransactionStatus;

public record PaymentCommand() {

	public record CreatePayment(
		String userId,
		String orderId,
		CardType cardType,
		String cardNo,
		Long amount,
		String callbackUrl
	) {
		public PaymentData.PaymentRequest toData() {
			return new PaymentData.PaymentRequest(userId, orderId, cardType, cardNo, amount, callbackUrl);
		}
	}

	public record ProcessCallback(
		String transactionKey,
		TransactionStatus status,
		String orderId
	) {
	}
}

File: apps/pg-simulator/src/main/kotlin/com/loopers/application/payment/PaymentCommand.kt
package com.loopers.application.payment

import com.loopers.domain.payment.CardType
import com.loopers.support.error.CoreException
import com.loopers.support.error.ErrorType

object PaymentCommand {
    data class CreateTransaction(
        val userId: String,
        val orderId: String,
        val cardType: CardType,
        val cardNo: String,
        val amount: Long,
        val callbackUrl: String,
    ) {
        fun validate() {
            if (amount <= 0L) {
                throw CoreException(ErrorType.BAD_REQUEST, "요청 금액은 0 보다 큰 정수여야 합니다.")
            }
        }
    }
}
